src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -83,13 +88,14 @@ class PeerManagerActor(
       val peerAddresses = managerState.outgoingPeers.values.map(_.remoteAddress).toSet
 
       val nodesToConnect = nodesInfo
-        .filterNot(n => peerAddresses.contains(n.node.tcpSocketAddress)) // not already connected to
-        .toSeq
-        .sortBy(-_.addTimestamp)
+        .filterNot(
+          n => peerAddresses.contains(n.node.tcpSocketAddress) || isBlacklisted(PeerAddress(n.node.tcpSocketAddress.getHostString))
```

Comment:
Yep i have also noticed that that's why I put a pretty low limit on this list.  I have run it with some profiling on and it would rather not be a problem in terms of performance.  
But after some thought (and your comment) I am gonna change it to `LinkedHashMap` as it could be a pretty good trap for the future, and next release is performance edition so let's keep it lean where we can, and this change wont is a big one.
ps. i also prefer prefer immutable on "var"s than mutable over "val"s, but in terms of performance mutable collections are usually faster generally speaking